src/shallow.ts

```diff
@@ -1,12 +1,7 @@
-function shallow<T extends (...args: any[]) => any>(objA: T, objB: T): boolean
-
-function shallow<T extends string | number | boolean>(objA: T, objB: T): boolean
-
-function shallow<T extends any[]>(objA: T, objB: T): boolean
-
-function shallow<T extends Record<string, any>>(objA: T, objB: T): boolean
-
-function shallow<T>(objA: T, objB: T) {
+function shallow<T extends string | number | boolean | Record<string, any>>(
```

Comment:
Why not also add `unknown[]` and even `undefined | null | symbol | bigint`? `Object.is` can compare them afaik.

Reply:
I actually wanted to make it to accept only array and object, because it should cover 90% use cases.
Using `shallow` for string only is unnecessary. `string | number | boolean` is added to avoid breaking existing tests.

All that said, this doesn't work as I hoped in #1064, and I added a test a07a7d9, so I'm actually okay to revert it.

```suggestion
function shallow<T>(
```